tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -4385,19 +4385,17 @@ bool MIOpenSupport::DoPoolBackward(
   }
 
   DeviceMemory<uint8> dest2;  // duplicated dest from forward:
-  int dest2_size = 0;
+  int64 dest2_size = 0;
 
   // miopen requires the strides and dims to be ordered as BDYX.
   std::vector<int64> dims64 =
       output_dimensions.full_dims(dnn::DataLayout::kBatchDepthYX);
-
   // miopen does not use strides and must have 4D tensor.
-  std::vector<int> dims(4);
+  std::vector<int> dims(pooling_dimensions.ndims() + 2);
 
-  std::transform(dims64.cbegin(), dims64.cend(), dims.begin(),
-                 &CheckedNarrowing<int64, int>);
-
-  dest2_size = dims[0] * dims[1] * dims[2] * dims[3] * sizeof(float);
+  dest2_size = sizeof(float);
+  for(auto& x: dims64)
+     dest2_size *= x;
```

Comment:
Why not use the existing MIOpen API `wrap::miopenPoolingGetWorkSpaceSizeV2` for this? 

Does the API internally did the workspace size calculation as you do here? If so, I'd still prefer to invoke their API, for future compatibility.

Reply:
this is not the workspace size, this is the output size.

Reply:
Ha, that's right...